java/kafka/producer/src/main/java/KafkaProducerConfig.java

```diff
@@ -59,6 +78,19 @@ public KafkaProducerConfig(String bootstrapServers, String topic, int delay, Lon
     }
 
     public static KafkaProducerConfig fromEnv() {
+        String serviceName = System.getenv("OTEL_SERVICE_NAME")  == null ? DEFAULT_SERVICE_NAME : System.getenv("OTEL_SERVICE_NAME");
+        String tracesExporterNames = System.getenv("OTEL_TRACES_EXPORTER")  == null ? DEFAULT_EXPORTER_NAMES : System.getenv("OTEL_TRACES_EXPORTER");
+        String metricsExporterNames = System.getenv("OTEL_METRICS_EXPORTER")  == null ? DEFAULT_METRICS_NAMES : System.getenv("OTEL_METRICS_EXPORTER");
+
+        boolean openTracingEnabled = System.getenv("OPENTRACING_ENABLED")  == null ? DEFAULT_OPENTRACING_ENABLED : Boolean.parseBoolean(System.getenv("OPENTRACING_ENABLED"));
+        boolean openTelemetryEnabled = System.getenv("OPENTELEMETRY_ENABLED")  == null ? DEFAULT_OPENTELEMETRY_ENABLED : Boolean.parseBoolean(System.getenv("OPENTELEMETRY_ENABLED"));
```

Comment:
All my comments about "enabled" env vars and metrics I made for the consumer are valid here as well.